multicluster/test/e2e/framework.go

```diff
@@ -101,106 +79,130 @@ func (data *TestData) createClients() error {
+	if d, ok := data.clusterTestDataMap[clusterName]; ok {
+		if err := d.DeleteNamespace(namespace, timeout); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func (data *TestData) createPod(clusterName string, name string, namespace string, ctrName string, image string, command []string,
+func (data *MCTestData) createPod(clusterName string, name string, namespace string, ctrName string, image string, command []string,
 	args []string, env []corev1.EnvVar, ports []corev1.ContainerPort, hostNetwork bool, mutateFunc func(pod *corev1.Pod)) error {
-	client := data.getClientOfCluster(clusterName)
-	return createPod(client, name, namespace, ctrName, image, command, args, env, ports, hostNetwork, mutateFunc)
-}
-
-func (data *TestData) getService(clusterName string, namespace string, serviceName string) (*corev1.Service, error) {
-	client := data.getClientOfCluster(clusterName)
-	return getService(client, namespace, serviceName)
+	if d, ok := data.clusterTestDataMap[clusterName]; ok {
+		if err := d.CreatePodOnNodeInNamespace(name, namespace, "", ctrName, image, command, args, env, ports, hostNetwork, mutateFunc); err != nil {
+			return err
+		}
+	}
+	return nil
 }
 
-func (data *TestData) createService(cluster string, serviceName string, namespace string, port int32, targetPort int32,
+func (data *MCTestData) createService(clusterName string, serviceName string, namespace string, port int32, targetPort int32,
 	protocol corev1.Protocol, selector map[string]string, affinity bool, nodeLocalExternal bool, serviceType corev1.ServiceType,
 	ipFamily *corev1.IPFamily, annotation map[string]string) (*corev1.Service, error) {
-	client := data.getClientOfCluster(cluster)
-	return createService(client, serviceName, namespace, port, targetPort, protocol, selector, affinity, nodeLocalExternal, serviceType, ipFamily, annotation)
+	if d, ok := data.clusterTestDataMap[clusterName]; ok {
+		svc, err := d.CreateServiceWithAnnotations(serviceName, namespace, port, targetPort, protocol, selector, affinity, nodeLocalExternal, serviceType, ipFamily, annotation)
+		if err != nil {
+			return nil, err
+		}
+		return svc, nil
+	}
+	return nil, fmt.Errorf("clusterName %s not found", clusterName)
 }
 
-func (data *TestData) getClientOfCluster(clusterName string) kubernetes.Interface {
-	return data.clients[clusterName]
+func (data *MCTestData) createOrUpdateANP(clusterName string, anp *crdv1alpha1.NetworkPolicy) (*crdv1alpha1.NetworkPolicy, error) {
+	if d, ok := data.clusterTestDataMap[clusterName]; ok {
+		return d.CreateOrUpdateANP(anp)
+	}
+	return nil, fmt.Errorf("clusterName %s not found", clusterName)
 }
 
-func (data *TestData) getCRDClientOfCluster(clusterName string) crdclientset.Interface {
-	return data.crdClients[clusterName]
+// deleteANP is a convenience function for deleting ANP by name and Namespace.
+func (data *MCTestData) deleteANP(clusterName string, ns, name string) error {
```

Comment:
I suppose it can be `(clusterName, ns, name string)` if you merge the type in method signature.

Reply:
Changed